File: content/排序_选择排序/selectionSort.cpp
/*
选择排序
每次遍历时，选择最小的元素与数组开头的元素进行交换
时间复杂度为O(n^2)，交换次数为 n 
*/

#include<iostream>
#include<vector>
using namespace std;


void selectionSort(vector<int> &array){
  for (int j = 0; j < array.size();j++)
  {
    int min = j;
    for (int i = j; i < array.size();i++)
    {
      if (array[i] < array[min])
      {
        min = i;
      }
    }
    swap(array[min], array[j]);
  }
}

int main(){
  vector<int> a = {2, 1, 3, 4, 7, 8, 6, 5};
  
  selectionSort(a);
  for (auto i:a){
    cout << i<< endl;
  }
  return 0;
}

File: content/selectionSort/SelectionSort.java
package content.selectionSort;

public class SelectionSort{
  public static void selectSort(int[] list) {
    for (int i = 0; i < list.length; i++) {
      int min_index = i;
      for (int j = i; j < list.length; j++) {
        if (list[j] < list[min_index]) {
          min_index = j;
        }
      }
      swap(list, i, min_index);
    }
  }
  
  public static void swap(int[] list, int i, int j) {
    int temp = list[i];
    list[i] = list[j];
    list[j] = temp;
  }

  public static void main(String[] args) {
    int[] list = { 1, 4, 2, 3, 5, 9, 7 };
    selectSort(list);
    for (int i : list) {
      System.out.println(i);
    }
  }
}